WMFComponents/Sources/WMFComponents/Extensions/Navigation Bar/WMFNavigationBarConfiguring.swift

```diff
@@ -164,27 +183,45 @@ public extension WMFNavigationBarConfiguring where Self: UIViewController {
             navigationItem.titleView = nil
         }
         
-        // Setup profile button if needed
+        var rightBarButtonItems: [UIBarButtonItem] = []
+
         if let profileButtonConfig {
             let image = profileButtonImage(theme: WMFAppEnvironment.current.theme, needsBadge: profileButtonConfig.needsBadge)
-            
             let profileButton = UIBarButtonItem(image: image, style: .plain, target: profileButtonConfig.target, action: profileButtonConfig.action)
             
             profileButton.accessibilityLabel = profileButtonAccessibilityStrings(config: profileButtonConfig)
             profileButton.accessibilityIdentifier = profileButtonAccessibilityID
             
-            var rightBarButtonItems: [UIBarButtonItem] = [profileButton]
-            if let leadingBarButtonItem = profileButtonConfig.leadingBarButtonItem {
-                rightBarButtonItems.append(leadingBarButtonItem)
+            if let trailing = profileButtonConfig.trailingBarButtonItem {
+                rightBarButtonItems.append(trailing)
             }
+
+            rightBarButtonItems.append(profileButton)
             
-            if let trailingBarButtonItem = profileButtonConfig.trailingBarButtonItem {
-                rightBarButtonItems.insert(trailingBarButtonItem, at: 0)
+            if let leading = profileButtonConfig.leadingBarButtonItem {
+                rightBarButtonItems.append(leading)
             }
-            
-            navigationItem.rightBarButtonItems = rightBarButtonItems
         }
         
+        if let dataController = try? ArticleTabsDataController() {
```

Comment:
You might need to adjust the order that you are building up `rightBarButtonItems` here. I notice that the ordering is off for nav bars that need a variable leading button (Places, Saved, History):

Mock:
<img width="386" alt="Screenshot 2025-04-30 at 4 22 55 PM" src="https://github.com/user-attachments/assets/b5075dea-4f3a-4b19-a195-5fdfc3f989f8" />

PR (Filter should precede tabs):
<img width="565" alt="Screenshot 2025-04-30 at 4 22 25 PM" src="https://github.com/user-attachments/assets/98ad8067-a898-44a9-ad76-2983211e8666" />